app/assets/javascripts/discourse/routes/build-category-route.js.es6

```diff
@@ -63,9 +63,15 @@ export default (filter, params) => {
 
     setupController(controller, model) {
       const topics = this.get('topics'),
-            periodId = topics.get('for_period') || (filter.indexOf('/') > 0 ? filter.split('/')[1] : '');
+            periodId = topics.get('for_period') || (filter.indexOf('/') > 0 ? filter.split('/')[1] : ''),
+            canCreateTopic = topics.get('can_create_topic'),
+            canCreateTopicOnCategory = model.get('permission') === Discourse.PermissionType.FULL;
 
-      this.controllerFor('navigation/category').set('canCreateTopic', topics.get('can_create_topic'));
+      this.controllerFor('navigation/category').setProperties({
+        canCreateTopicOnCategory: canCreateTopicOnCategory,
+        cannotCreateTopicOnCategory: !canCreateTopicOnCategory,
```

Comment:
Is there an advantage to using the computed property? I was leaning towards
this implementation since it puts all the related code in one place.
On Sep 18, 2015 8:06 PM, "Kane York" <notifications@github.com> wrote:

> In app/assets/javascripts/discourse/routes/build-category-route.js.es6
> <https://github.com/discourse/discourse/pull/3788#discussion_r39912305>:
>
> >
> > -      this.controllerFor('navigation/category').set('canCreateTopic', topics.get('can_create_topic'));
> > +      this.controllerFor('navigation/category').setProperties({
> > +        canCreateTopicOnCategory: canCreateTopicOnCategory,
> > +        cannotCreateTopicOnCategory: !canCreateTopicOnCategory,
>
> You can add a property to controller/navigation/category.js.es6 like this:
>
> cannotCreateTopicOnCategory: Ember.computed.not('canCreateTopicOnCategory'),
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/discourse/discourse/pull/3788/files#r39912305>.
>